### PR TITLE
Handle Option inside containers

### LINKED ISSUE #61: Containers with optional values are encoded incorrectly
## description
See #58

### PR DESCRIPTION
**Pull request by gabro**
Closes #61 

This PR adds compatibility for handling `Option` when it appears inside containers.

Suppose we have this case class in Scala:

```scala
case class Foo(
  bar: Option[Int],
  baz: List[Option[Int]]
)
```

Today, metarpheus-io-ts produces this interface:

```typescript
interface Foo {
  bar?: number,
  baz: Array<number>
}

const Foo = t.interface({
  bar: t.union([
    t.number,
    t.undefined
  ]),
  baz: t.array(t.number)
})
```

We're then missing the optionality of whatever lives inside a container.

This PR makes metarpheus-io-ts to produce this interface instead:

```typescript
interface Foo {
  bar?: number,
  baz: Array<
   | number
   | undefined
  >
}

const Foo = t.interface({
  bar: t.union([
    t.number,
    t.undefined
  ]),
  baz: t.array(t.union([
    t.number,
    t.undefined
  )]
})
```

We still handle "plain" optional values as before, but we use a union type in case the optional type lives inside a container.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -50,7 +50,7 @@ function genericCombinator(tpe: Tpe): Reader<Ctx, gen.CustomCombinator> {
   });
 }
 
-export function getType(tpe: Tpe): Reader<Ctx, gen.TypeReference> {
+export function getType(tpe: Tpe, owner: Tpe | null): Reader<Ctx, gen.TypeReference> {
```
Comment (gabro) [inline src/index.ts:53]: I could make the `owner` parameter optional, but I want make sure it can't be forgotten.

**Comment by gabro** [comment]
Removed yarn.lock

**Review by giogonzo — APPROVED** [review]
LGTM 👍I would still prefer to have a more general solution for this, but it could be something for the future

**Comment by gabro** [comment]
Yeah, I'll rethink this to also include the "partials" encoding for optional type.

I've switched from `undefined` to `null` since it's impossible for Scala to serialize an `Array<undefined>` (`undefined` does not exist in JSON).

**Pull request merged**